packages/fieldset/test/lion-fieldset.test.js

```diff
@@ -342,66 +332,151 @@ describe('<lion-fieldset>', () => {
+
+    it('[deprecated] sets a class "state-(touched|dirty)"', async () => {
+      const el = await fixture(html`<${tag}></${tag}>`);
+      el.touched = true;
+      await el.updateComplete;
+      expect(el.classList.contains('state-touched')).to.equal(true, 'has class "state-touched"');
+
+      el.dirty = true;
+      await el.updateComplete;
+      expect(el.classList.contains('state-dirty')).to.equal(true, 'has class "state-dirty"');
+    });
+
+    it('becomes prefilled if all form elements are prefilled', async () => {
+      const el = await fixture(html`
+        <${tag}>
+          <${childTag} name="input1" prefilled></${childTag}>
+          <${childTag} name="input2"></${childTag}>
+        </${tag}>
+      `);
+      await nextFrame();
+      expect(el.prefilled).to.be.false;
+
+      const el2 = await fixture(html`
+        <${tag}>
+          <${childTag} name="input1" prefilled></${childTag}>
+          <${childTag} name="input2" prefilled></${childTag}>
+        </${tag}>
+      `);
       await nextFrame();
-      await triggerFocusFor(fieldset.formElements['gender[]'][0].inputElement);
-      await triggerBlurFor(fieldset.formElements['gender[]'][0].inputElement);
-      expect(fieldset.touched).to.equal(true);
+      expect(el2.prefilled).to.be.true;
     });
 
-    it('sets a class "state-(touched|dirty)"', async () => {
-      const fieldset = await fixture(html`<${tag}>${inputSlots}</${tag}>`);
+    it(`becomes "touched" once the last element of a group becomes blurred by keyboard
+      interaction (e.g. tabbing through the checkbox-group)`, async () => {
+      const el = await fixture(html`
+        <${tag}>
+          <label slot="label">My group</label>
+          <${childTag} name="myGroup[]" label="Option 1" value="1"></${childTag}>
+          <${childTag} name="myGroup[]" label="Option 2" value="2"></${childTag}>
+        </${tag}>
+      `);
       await nextFrame();
-      fieldset.formElements.color.touched = true;
-      await fieldset.updateComplete;
-      expect(fieldset.classList.contains('state-touched')).to.equal(
+
+      const button = await fixture(`<button>Blur</button>`);
+
+      expect(el.touched).to.equal(false, 'initially, touched state is false');
+      el.children[2].focus();
+      expect(el.touched).to.equal(false, 'focus is on second checkbox');
+      button.focus();
+      expect(el.touched).to.equal(
         true,
-        'has class "state-touched"',
+        `focus is on element behind second checkbox (group has blurred)`,
       );
-
-      fieldset.formElements.color.dirty = true;
-      await fieldset.updateComplete;
-      expect(fieldset.classList.contains('state-dirty')).to.equal(true, 'has class "state-dirty"');
     });
 
-    it('sets prefilled when field left and value non-empty', async () => {
-      const fieldset = await fixture(html`<${tag}>${inputSlots}</${tag}>`);
+    it(`becomes "touched" once the group as a whole becomes blurred via mouse interaction after
```

Comment:
improve this by testing multiple fields not just one.

Reply:
done